data_adapter_oemof/build_datapackage.py

```diff
@@ -1,92 +1,185 @@
+        for (sequence_name, sequence_dataframe) in parametrized_sequences.items():
+            # Group the DataFrame by year using pd.Grouper
+            grouped = sequence_dataframe.resample("Y")
+
+            # Iterate over the groups and store each year's DataFrame
+            for year, group in grouped:
+                split_dataframes[sequence_name+"_"+str(year.year)] = group.copy()
+
+        return split_dataframes
+
+    def get_foreign_keys(self):
+        pass
 
+    def save_datapackage_to_csv(self, destination):
+        """
+        Saving the datapackage to a given destination in oemof.tabular readable format
+        :param destination:
+        :return:
         """
 
     @classmethod
     def build_datapackage(
-        cls, es_structure, **process_data: dict[str, preprocessing.Process]
+        cls, adapter: Adapter
     ):
         """
-        Adapting the resource scalars for a Datapackage using adapters from adapters.py
+        Adapting the resource scalars for a Datapackage using adapters from
+        adapters.py
+        :type adapter: Adapter class from preprocessing
+        :return: Instance of datapackage class
 
-        :param es_structure:
-        :param process_data: preprocessing.Process
-        :return:
         """
-        # Initializing empty dict for scalar data:
+        es_structure = adapter.get_structure()
         parametrized_elements = {}
-
-        # Initializing empty dict for timeseries data:
         parametrized_sequences = {}
-
-        for (process, data) in process_data.items():
-            # Getting Process parameters:
-            process_type: str = PROCESS_TYPE_MAP[process]
-            adapter = TYPE_MAP[process_type]
-
-            # Apply Adapter to every instance of the Process:
-            scalars = data.scalars.apply(
-                adapter.parametrize_dataclass,
-                struct=es_structure[process],
-                process_type=process_type,
+        foreign_keys = {}
+        for (process_name, struct) in es_structure.items():
+            process = adapter.get_process(process_name)
+            facade_adapter_name: str = PROCESS_TYPE_MAP[process_name]
+            facade_adapter = TYPE_MAP[facade_adapter_name]
+
+            scalars = process.scalars.apply(
+                func=facade_adapter.parametrize_dataclass,
+                struct=struct,
                 axis=1,
             )
-            # Maybe work on timeseries here:
-            sequences = data.timeseries
-
-            # Add process to parametrized_elements dict:
-            if process_type in parametrized_elements.keys():
-                parametrized_elements[process_type] = pd.concat(
-                    [
-                        pd.DataFrame([param.as_dict() for param in scalars.values]),
-                        parametrized_elements[process_type],
-                    ],
+
+            scalars = pd.DataFrame(scalars.to_list())
+            if "profiles" in facade_adapter.__dict__:
```

Comment:
I think refactoring timeseries can be more generalized and instead mapping of profiles to TS columns should be added - will tell you my idea in a meeting.
Then this hardcoded line could be removed and limiting TS to single column could be removed as well